### ISSUE TITLE
seg fault in IsMaistra status

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#4351]
```
2021-09-14T16:26:03Z INF Kiali cache for [namespace: istio-system] started
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x140e489]

goroutine 103 [running]:
github.com/kiali/kiali/status.IsMaistra(...)
	/home/jmazzite/source/kiali/kiali/src/github.com/kiali/kiali/status/status.go:114
```

**Comment by jmazzitelli** [comment]
Looks like its possible the status hasn't been obtained prior to this IsMaistra being called. We need to protect for that.

**Comment by jmazzitelli** [comment]
In case we want to fix this now and build a 1.40.1, here's PRs that need to go in first - then we can kick off a patch build:

* https://github.com/kiali/kiali/pull/4353
* https://github.com/kiali/kiali-operator/pull/423
* https://github.com/kiali/kiali-operator/pull/424

**Comment by a2659802** [comment]
do we meet the same situation?

```
2021-12-21T08:20:13Z INF Kiali: Version: v1.40.0, Commit: 9ad4aff742ce58462d03701d51da0e70360a436c
2021-12-21T08:20:13Z INF Using authentication strategy [anonymous]
2021-12-21T08:20:13Z WRN Kiali auth strategy is configured for anonymous access - users will not be authenticated.
2021-12-21T08:20:13Z INF Kiali: Console version: 1.40.0
2021-12-21T08:20:13Z INF Server endpoint will start at [:20001/kiali]
2021-12-21T08:20:13Z INF Server endpoint will serve static content from [/opt/kiali/console]
2021-12-21T08:20:13Z INF Starting Metrics Server on [:9090]
2021-12-21T08:20:25Z INF Kiali Cache is active for namespaces [.*]
2021-12-21T08:20:25Z INF [Prom Cache] Enabled
2021-12-21T08:20:25Z INF Waiting for Kiali cache for [namespace: istio-system] to sync
2021-12-21T08:20:25Z INF Kiali cache for [namespace: istio-system] started
2021-12-21T08:20:25Z INF Waiting for Kiali cache for [namespace: tmp] to sync
2021-12-21T08:20:25Z INF Kiali cache for [namespace: tmp] started
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x140e2a9]
goroutine 113 [running]:
github.com/kiali/kiali/status.IsMaistra(...)
/home/fedora/workspace/kiali-workflow/kiali-release/src/github.com/kiali/kiali/status/status.go:114
github.com/kiali/kiali/models.(*Workload).parseObjectMeta(0xc0002c6c60, 0xc000278d88, 0xc000278e90)
/home/fedora/workspace/kiali-workflow/kiali-release/src/github.com/kiali/kiali/models/workload.go:172 +0x1c9
github.com/kiali/kiali/models.(*Workload).ParseDeployment(0xc0002c6c60, 0xc000278d68)
/home/fedora/workspace/kiali-workflow/kiali-release/src/github.com/kiali/kiali/models/workload.go:202 +0x67
github.com/kiali/kiali/business.fetchWorkloads(0xc000429680, 0xc000553170, 0xc, 0x0, 0x0, 0x20063521a, 0x1000200000001, 0xc0003229c0, 0xc0004541e0, 0xc00004b500)
/home/fedora/workspace/kiali-workflow/kiali-release/src/github.com/kiali/kiali/business/workloads.go:811 +0x340f
github.com/kiali/kiali/business.(*IstioStatusService).getComponentNamespacesWorkloads.func1(0xc000134170, 0xc0004296e8, 0xc000553170, 0xc, 0xc00061e1e0, 0xc00061e240)
/home/fedora/workspace/kiali-workflow/kiali-release/src/github.com/kiali/kiali/business/istio_status.go:112 +0x9f
created by github.com/kiali/kiali/business.(*IstioStatusService).getComponentNamespacesWorkloads
/home/fedora/workspace/kiali-workflow/kiali-release/src/github.com/kiali/kiali/business/istio_status.go:108 +0x4ed
```

**Comment by jmazzitelli** [comment]
> do we meet the same situation?

yes.

**Comment by adabuleanu** [comment]
In what version was this fixed? I have 1.58.0 and get this error
```
{"level":"trace","time":"2023-01-18T12:02:21Z","message":"[Kiali Cache] Get [resource: Deployment] for [namespace: istio-system] [name: istiod]"}
--
2023/01/18 12:02:21 http: panic serving 10.0.14.1:15037: runtime error: invalid memory address or nil pointer dereference
goroutine 879 [running]:
net/http.(*conn).serve.func1()
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:1802 +0xb9
panic({0x1b0cd40, 0x3066510})
/opt/hostedtoolcache/go/1.17.10/x64/src/runtime/panic.go:1047 +0x266
github.com/kiali/kiali/business.(*MeshService).IstiodResourceThresholds(0xc004774318)
/home/runner/work/kiali/kiali/business/mesh.go:638 +0x119
github.com/kiali/kiali/handlers.IstiodResourceThresholds({0x2095280, 0xc001e3aba0}, 0xc001ba6ff0)
/home/runner/work/kiali/kiali/handlers/mesh.go:43 +0x90
net/http.HandlerFunc.ServeHTTP(0x1de8079, {0x2095280, 0xc001e3aba0}, 0x7f7a9bcddf18)
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:2047 +0x2f
github.com/kiali/kiali/routing.metricHandler.func1({0x20a2a20, 0xc0026617a0}, 0x2060501)
/home/runner/work/kiali/kiali/routing/router.go:145 +0x144
net/http.HandlerFunc.ServeHTTP(0x20aaed8, {0x20a2a20, 0xc0026617a0}, 0x2060510)
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:2047 +0x2f
github.com/kiali/kiali/handlers.AuthenticationHandler.Handle.func1({0x20a2a20, 0xc0026617a0}, 0xc00475ec00)
/home/runner/work/kiali/kiali/handlers/authentication.go:79 +0x514
net/http.HandlerFunc.ServeHTTP(0x1a1c460, {0x20a2a20, 0xc0026617a0}, 0xc0006ae280)
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:2047 +0x2f
github.com/kiali/kiali/server.plainHttpMiddleware.func1({0x20a2a20, 0xc0026617a0}, 0xc001ba6f90)
/home/runner/work/kiali/kiali/server/server.go:152 +0x65
net/http.HandlerFunc.ServeHTTP(0xc00475eb00, {0x20a2a20, 0xc0026617a0}, 0xc)
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:2047 +0x2f
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001629c0, {0x20a2a20, 0xc0026617a0}, 0xc00475ea00)
/home/runner/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1({0x20a28d0, 0xc003930b60}, 0xea15dad9150102)
/home/runner/go/pkg/mod/github.com/!n!y!times/gziphandler@v1.1.1/gzip.go:336 +0x23c
net/http.HandlerFunc.ServeHTTP(0x0, {0x20a28d0, 0xc003930b60}, 0x0)
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:2047 +0x2f
net/http.(*ServeMux).ServeHTTP(0x0, {0x20a28d0, 0xc003930b60}, 0xc00475ea00)
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:2425 +0x149
net/http.serverHandler.ServeHTTP({0xc001ba6d20}, {0x20a28d0, 0xc003930b60}, 0xc00475ea00)
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:2879 +0x43b
net/http.(*conn).serve(0xc003b11b80, {0x20aaed8, 0xc003ccb5c0})
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:1930 +0xb08
created by net/http.(*Server).Serve
/opt/hostedtoolcache/go/1.17.10/x64/src/net/http/server.go:3034 +0x4e8
```

**Comment by jmazzitelli** [comment]
> In what version was this fixed? I have 1.58.0 and get this error

That stack trace does not involve the isMaistra function or this github issue. Create a different github issue for this. That stack trace does not look familiar to me. 

When you write a new github issue, please indicate which versions of both Kiali AND istio you are using. In the past, we have seen errors like this when people are using the wrong version of Kiali for the Istio version in their environment. Please confirm with the compatibility matrix here: https://kiali.io/docs/installation/installation-guide/prerequisites/#version-compatibility
